### PR TITLE
Report interactor compile error as internal error

### PR DESCRIPTION
**Pull request by Balint-R**
Before this change, the interactor failing to compile was reported in the same way as the user's submission failing to compile.

### CODE REVIEW

File: dmoj/graders/bridged.py [hunk 1]
```diff
@@ -14,9 +15,14 @@ class BridgedInteractiveGrader(StandardGrader):
     def __init__(self, judge, problem, language, source):
         super().__init__(judge, problem, language, source)
         self.handler_data = self.problem.config.interactive
-        self.interactor_binary = self._generate_interactor_binary()
-        self.contrib_type = self.handler_data.get('type', 'default')
 
+        try:
+            self.interactor_binary = self._generate_interactor_binary()
+        except CompileError as compilation_error:
+            # Rethrow as IE to differentiate from the user's submission CEing
```
Comment (kiritofeng) [inline dmoj/graders/bridged.py:22]:
```suggestion
            # Rethrow as IE to differentiate from the user's submission failing to compile.
```

File: dmoj/graders/bridged.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import subprocess
 
 from dmoj.contrib import contrib_modules
+from dmoj.error import CompileError
```
Comment (kiritofeng) [inline dmoj/graders/bridged.py:6]: Can we merge this with the line below (so `from dmoj.error import CompileError, InternalError`

**Review by kiritofeng — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Review by kiritofeng — APPROVED** [review]
LGTM, thanks!

File: dmoj/graders/bridged.py [hunk 3]
```diff
@@ -14,9 +14,14 @@ class BridgedInteractiveGrader(StandardGrader):
...
         super().__init__(judge, problem, language, source)
         self.handler_data = self.problem.config.interactive
-        self.interactor_binary = self._generate_interactor_binary()
-        self.contrib_type = self.handler_data.get('type', 'default')
 
+        try:
+            self.interactor_binary = self._generate_interactor_binary()
+        except CompileError as compilation_error:
+            # Rethrow as IE to differentiate from the user's submission failing to compile.
+            raise InternalError("Interactor failed compiling") from compilation_error
```
Comment (kiritofeng) [inline dmoj/graders/bridged.py:22]: Can you change these to single quotes so flake8 stops being sad

File: dmoj/graders/bridged.py [hunk 4]
```diff
@@ -14,9 +14,14 @@ class BridgedInteractiveGrader(StandardGrader):
...
         super().__init__(judge, problem, language, source)
         self.handler_data = self.problem.config.interactive
-        self.interactor_binary = self._generate_interactor_binary()
-        self.contrib_type = self.handler_data.get('type', 'default')
 
+        try:
+            self.interactor_binary = self._generate_interactor_binary()
+        except CompileError as compilation_error:
+            # Rethrow as IE to differentiate from the user's submission failing to compile.
+            raise InternalError('Interactor failed compiling') from compilation_error
```
Comment (Xyene) [inline dmoj/graders/bridged.py:22]: Nit for consistency with most other errors
```suggestion
            raise InternalError('interactor failed compiling') from compilation_error
```

**Review by Xyene — APPROVED** [review]
LGTM, thanks!

**Pull request merged**